PyInstaller/compat.py

```diff
@@ -73,6 +48,29 @@
 is_py312 = sys.version_info >= (3, 12)
 is_py313 = sys.version_info >= (3, 13)
 
+# PyInstaller requires importlib.metadata from python >= 3.10 stdlib, or equivalent importlib-metadata >= 4.6.
+if not _setup_py_mode:
+    if not is_py310:
+        try:
+            vendored = importlib.import_module("importlib_metadata")
+            if parse_version(vendored.version("importlib-metadata")) < parse_version("4.6"):
+                raise ImportError("importlib-metadata must be version 4.6 or higher")
+
+            sys.modules["importlib.metadata"] = vendored
```

Comment:
Doesn't this mean that during analysis, `importlib.metadata` will also resolve into `importlib_metadata` instead of the stdlib version?

Reply:
Yes, but only on python 3.8 & 3.9, where `importlib.metadata` does not exist. I don't particularly see a way this should cause any issues?

Reply:
If user's application tries to import stdlib version, it should get stdlib version. The installed `importlib_metadata` (which PyInstaller requires) might be of significantly newer version, and might have an incompatible API.

Reply:
Why do we require `importlib_metadata >= 4.6` for 3.8 and 3.9 then? Is there a feature we require that's not in the standard library version for those python versions?

Reply:
> Why do we require `importlib_metadata >= 4.6` for 3.8 and 3.9 then? Is there a feature we require that's not in the standard library version for those python versions?

Yeah, we need the version from 3.10 stdlib or (equivalent) importlib-metadata >= 4.6 for `group` and `name` keywords in the `entry_points()` function.